enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,338 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This EP describes an addition of the new special platform type, referred here as "External".
+
+Such new platform type would serve as a signal when OpenShift is running on an infrastructure platform
+that will receive supplemental components from the infrastructure provider.
+
+“External” is different from our current platform types in that it will signal that the infrastructure
+is specifically not “None” or any of the known providers (eg AWS, GCP, etc).
+
+## Motivation
+
+At the moment [Infrastructure](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L14)
+resource serves as a main source of information about the underlying infrastructure provider and provider specific parameters.
+Specifically [PlatformSpec](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L47)
+and [PlatformStatus](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L65) parts
+of the infrastructure resources holds that data.
+
+Given that PlatformSpec and PlatformStatus defined as a "discriminated unions" and require to have platform type encoded within
+"openshift/api" beforehand, it creates significant footprint and plenty of effort from RedHat engineers to initial technical enablement of
+a new cloud-provider and mostly impossible without RedHat engineering engagement.
+
+Since a lot of infrastructure related components (such as [CCCMO](https://github.com/openshift/cluster-cloud-controller-manager-operator), [Machine API operator](https://github.com/openshift/machine-api-operator), [Machine Config Operator](https://github.com/openshift/machine-config-operator), and so on)
+requires an information about, at least, presence of underlying cloud-provider - "None" platform does not suite well as a signal in such situation.
+
+Having special, built-in and somewhat generic, platform-type which will signal about presence of underlying infrastructure platform without
+platform specific details might help to reduce a number of pull-requests across OCP repositories
+and simplify initial integration work for external to RedHat contributors.
+
+In the future, at some degree, [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+API might help to solve the issue and give a room for supplementing *some* platform dependant components, however some integral parts of 
+OpenShift can not be disabled and still requires a signal about underlying platform, for example KCM and MCO in respect of enablement of
+[external cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/)
```

Comment:
i think the example is great here, but we should be more definitive about how we want to use the capabilities. we will use external platform in combination with capabilities to know how our components will react on these style of deployments.

a great example here of how we will use capabilities with the platform type is the MAO and how we want to run the generic controllers. if platform "external" is specified and the "machine api" capability is enabled, then the MAO will know what to do.